### ISSUE TITLE
default logging

### ISSUE THREAD

**Issue by mabe71** [drohm/pi-sht1x#10]
Dear Doug,

thanks for pi-sht1x. I'm using version 1.09 and it works really well except for the default logging as it is defined in the code.

I'm using the library to read the data from several sensors every couple of minutes. This means with logging level INFO a lot of data is produced. In fact, the log runs over about every 6 hours. I changed this by setting the logging level to WARNING. Now, I get much smaller logs. I gues that most users do not need such extensive logs and would therefore propose that you change the default logging level to something higher than INFO when you make an update to the library.

Moreover, GPIO access does not require root privileges any more. In a central installation, pi-sht1x still requires root privileges because of the location of the log. Moving the log elsewhere would make root priviledges not necessary anymore. It works well for me with ordinary user priviledges after I changed the write permissions to the pi-sht1x directory. In my case this is no problem because it is a quasi single user system. But this is not an ideal solution. Maybe you want to consider this point, too.

Regards,
Matt.

**Comment by drohm** [comment]
Hi Matt,
Thanks for using the library and for the suggestions.

I agree, both items you bring up should be adjusted.  When I wrote the library root was required but now that it's not required anymore I need to move the log destination.  The log verbosity should also be WARNING by default.  I'll get these updated in the next release.

Doug

**Comment by drohm** [comment]
One way to get around the logging directory is to pass in a logger to the SHT1x class.  Take a look at the logging.py script.  Simply create your own logger and set the log file path accordingly.

While you're creating the logger, you can set the default logging level as well.

Give that a go and let me know.

**Comment by mabe71** [comment]
Well, my knowledge of python and oop is rather limited. Nevertheless I tried. 

I copied essentially all of logging.py into my script. I don't know if this is necessary or if I needed only part of it.

In my main() I added
`mylogger = create_logger(__name__)`
and I called the sensor with the additional parameter:
`with SHT1x(DATA_PIN, SCK_PIN, gpio_mode=GPIO.BOARD, logger=mylogger) as sensor:`

Now, the logfile is where I defined it in my definition of the logger and also with the log level that I defined there. Both different from the default in the library.

So, this seems to solve the problem.

Thanks.
Matt
